UnitTests/CommonTestUtils/AsyncTestHelper.cs

```diff
@@ -34,10 +35,14 @@ public static T RunAndWaitForPendingOperations<T>(Func<Task<T>> asyncMethod, Can
 
         public static void WaitForPendingOperations(TimeSpan timeout)
         {
+            // Workaround for tests hanging in conjunction with canceled operations:
+            // Process the message loop before and after the wait.
+            Application.DoEvents();
             using (var cts = new CancellationTokenSource(timeout))
             {
                 WaitForPendingOperations(cts.Token);
             }
+            Application.DoEvents();
```

Comment:
Without `Application.DoEvents`, the tests occasionally hang (until AppVeyor cancels them after 1 hour).